pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -118,13 +305,38 @@ func (conn *Connection) NewRequest(creds *auth.Credentials, pid uint32, ino uint
 }
 
 // Call makes a request to the server and blocks the invoking task until a
-// server responds with a response.
-// NOTE: If no task is provided then the Call will simply enqueue the request
-// and return a nil response. No blocking will happen in this case. Instead,
-// this is used to signify that the processing of this request will happen by
-// the kernel.Task that writes the response. See FUSE_INIT for such an
-// invocation.
+// server responds with a response. Task should never be nil.
+// For async tasks, use CallAsync().
 func (conn *Connection) Call(t *kernel.Task, r *Request) (*Response, error) {
+	// Block requests sent before connection is initalized.
+	if !conn.IsInitialized() {
+		if err := t.Block(conn.initializedChan); err != nil {
+			return nil, err
+		}
+	}
+
+	return conn.callDo(t, r)
+}
+
+// CallAsync makes an async (aka background) request.
+// Those requests either do not expect a response (e.g. release) or
+// the response should be handled by others (e.g. init).
+// Return immediately unless the connection is blocked.
+// Async call example: init, release, forget, aio, interrupt.
+func (conn *Connection) CallAsync(t *kernel.Task, r *Request) error {
+	// Block requests sent before connection is initalized.
+	if !conn.IsInitialized() && r.hdr.Opcode != linux.FUSE_INIT {
+		if err := t.Block(conn.initializedChan); err != nil {
```

Comment:
Since `t` can be `nil`, the only reason this doesn't panic is because right now, `t` can only be `nil` when `conn.IsInitialized() == false`, which we check above. 

This is a pretty fragile way to prevent a panic due to nil pointer dereference. If `t` is allowed to be `nil`, please add an explicit nil check before dereferencing it:

```
if !conn.IsInitialized() && t != nil && r.hdr.OpCode != linux.FUSE_INIT {
    ...
}
```